modules/ztp-image-based-upgrade-with-rhacm.adoc

```diff
@@ -0,0 +1,362 @@
+---
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetwork
+metadata:
+  name: "networking-e5l"
+  namespace: openshift-sriov-network-operator
+spec:
+  ipam: |-
+    {
+    }
+  linkState: auto
+  networkNamespace: flexran-testcases
+  resourceName: pci_sriov_net_e5l
+  spoofChk: "on"
+  trust: "off"
+----
+
+.. Create the `ConfigMap` object:
++
+[source,terminal]
+----
+$ oc create configmap example-extra-manifests-cm --from-file=example-extra-manifests.yaml=<path_to_extramanifest> -n openshift-lifecycle-agent
+----
+
+. (Optional) To keep your custom catalog sources after the upgrade, generate another `ConfigMap` object for your catalog sources and add them to the `spec.extraManifest` field in the `ImageBasedUpgrade` CR. For more information about catalog sources, see xref:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.15/html-single/operators/index#olm-catalogsource_olm-understanding-olm[Catalog source].
+
+.. Create a YAML file that contains the `CatalogSource` CR.
++
+--
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  annotations:
+    target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+  name: example-catalogsources
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  displayName: disconnected-redhat-operators
+  image: quay.io/example-org/example-catalog:v1
+----
+--
+
+.. Create the `ConfigMap` object:
++
+[source,terminal]
+----
+$ oc create configmap example-catalogsources-cm --from-file=example-catalogsources.yaml=<path_to_catalogsource_cr> -n openshift-lifecycle-agent
+----
+
+. Edit the `ImageBasedUpgrade` CR:
++
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: ImageBasedUpgrade
+metadata:
+  name: example-upgrade
+spec:
+  stage: Idle
+  seedImageRef:
+    version: 4.15.2 <1>
+    image: <seed_container_image> <2>
+    pullSecretRef: <seed_pull_secret> <3>
+  autoRollbackOnFailure: {}
+#    initMonitorTimeoutSeconds: 1800 <4>
+  extraManifests: <5>
+  - name: example-extra-manifests-cm
+    namespace: openshift-lifecycle-agent
+  - name: example-catalogsources-cm
+    namespace: openshift-lifecycle-agent
+  oadpContent: <6>
+  - name: oadp-cm-example
+    namespace: openshift-adp
+----
+<1> Specify the target platform version. The value must match the version of the seed image.
+<2> Specify the repository where the target cluster can pull the seed image from.
+<3> Specify the reference to a secret with credentials to pull container images.
+<4> (Optional) Specify the time frame in seconds to roll back if the upgrade does not complete within that time frame after the first reboot. If not defined or set to `0`, the default value of `1800` seconds (30 minutes) is used.
+<5> (Optional) Specify the extra manifests to apply to the target cluster that are not part of the seed image. You can also add your custom catalog sources that you want to retain after the upgrade.
```

Comment:
IMHO "You can also add your custom catalog sources that you want to retain after the upgrade" suggests the catalog sources aren't part of extra manifests.

Reply:
What do you think about rephrasing like this?

> Specify the list of `ConfigMap` resources that contain your custom catalog sources to retain after the upgrade, and your extra manifests to apply to the target cluster that are not part of the seed image.